### PR TITLE
Return raw logits when using custom objective functions

### PR DESCRIPTION
**Pull request by rschmukler**
Continues the custom objective work of the previous release with returning the logits from the `predict` function if we have a custom objective. If you think about trying to implement an objective like `multiclass:softmax`, you would need the raw logits to be able to convert them to their final values.

### CODE REVIEW

**Comment by behrica** [comment]
Thanks for the PR.
Could you add one or  modify one of the existing tests, so it is using this new feature ?

**Comment by rschmukler** [comment]
The code path is technically covered by the existing custom objective test (https://github.com/scicloj/scicloj.ml.xgboost/blob/main/test/scicloj/ml/xgboost_test.clj#L425) - It's somewhat meaningless to to assert on the values of the raw logits themselves, but again, the existing test does ensure the code path is covered. If that isn't sufficient let me know what might be a better test and I can try and implement something

**Comment by behrica** [comment]
ok, I trust you on this. If a test adds no added value, we don't need it.

I will merge it.
